sim-lib/src/lib.rs

```diff
@@ -819,32 +838,32 @@ async fn produce_events<N: DestinationGenerator + ?Sized, A: PaymentGenerator +
                 let amount = match node_generator.payment_amount(capacity) {
                     Ok(amt) => {
                         if amt == 0 {
-                            log::debug!("Skipping zero amount payment for {source} -> {destination}.");
+                            log::debug!("{activity_name} activity => Skipping zero amount payment for {source} -> {destination}.");
```

Comment:
Ok. I'll replace that. 

And if a name isn't set, it should print the `index` of the `activity`. But I'll confirm this is true for this as well.

Reply:
We don't set an `activity_name` for randomly generated activity (above, line 776) right?

Reply:
Yes, we don't set an `activity_name` for randomly generated activity. I see your point on this logging ` activity` if a name isn't set for it.

How best do you think we can handle this? I had a name set for it before but you earlier mentioned we should take it off.

Reply:
By only logging the prefix if we've got a value there?

`if self.activity_name != "" {  format!("{} activity:", self.activity_name) } else {"".to_string()}`

Could inline, pulled out into a impl of `Payment`, or leave `activity_name` as an option so that we can use some of the `uwrap_or` functionality.

Reply:
Thank you for the clarification. I've addressed this and updated the PR.